artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -1078,21 +1088,26 @@ public synchronized MessageReference removeReferenceByID(final long messageID) t
          // This is an optimization, if the reference is the first one, we just poll it.
          // But first we need to make sure deliveringRefs isn't empty
          if (deliveringRefs.isEmpty()) {
+            if (logger.isTraceEnabled()) {
+               logger.trace("removeReferenceByID {} return null", messageID);
+            }
```

Comment:
Seems odd to gate this one, and not gate the few similar ones added in the next 20 lines below.